pkg/cache/tas_flavor_snapshot.go

```diff
@@ -803,22 +910,29 @@ func (s *TASFlavorSnapshot) sortedDomains(domains []*domain, unconstrained bool)
 	isLeastFreeCapacity := useLeastFreeCapacityAlgorithm(unconstrained)
 	result := slices.Clone(domains)
 	slices.SortFunc(result, func(a, b *domain) int {
```

Comment:
Can we add a comment how the sorting works?

Reply:
Sure! If you don't mind, I'll that improvement as an item for a follow up PR.